src/Microsoft.Azure.SignalR.Protocols/ServiceMessage.cs

```diff
@@ -126,6 +126,50 @@ public interface IAckableMessage
         int AckId { get; set; }
     }
 
+    /// <summary>
+    /// The ASRS send it to the source server connection to initiate a client connection migration.
+    /// Indicates that incoming messages are blocked.
+    /// The app server should block outgoing messages and reply with a <see cref="MigrateConnectionAckMessage"/>.
+    /// </summary>
+    public class MigrateConnectionRequestMessage : ConnectionMessage
```

Comment:
So this message could also be shared by every migration scenario

Reply:
How about... a `MigrateConnectionMessage` with state: `Requested` `Accepted` `Started`

Reply:
what about `Pause`, `PauseAck`, `Resume`

Reply:
and we can also use pausing and resuming to control traffic for streaming

Reply:
updated to `OutgoingTaskControlMessage` with `Pause`, `PauseAck` and `Resume` state.